static/js/containers/App.js

```diff
@@ -88,17 +89,14 @@ class App extends React.Component {
       location: { pathname },
       dispatch,
     } = this.props;
-    let [ complete, info ] = validateProfileComplete(profile);
+    let [ complete, url, errors] = validateProfileComplete(profile);
     if (
       userProfile.getStatus === FETCH_SUCCESS &&
       profile.agreed_to_terms_of_service &&
       !PROFILE_REGEX.test(pathname) &&
       !complete
     ) {
-      const { url, title, text } = info;
-      dispatch(updateDialogText(text));
-      dispatch(updateDialogTitle(title));
-      dispatch(setDialogVisibility(true));
+      dispatch(updateProfileValidation(errors));
```

Comment:
all you need to do is add

```js
import { startProfileEdit } from '../actions/index';
```
up top, and do

```js
  dispatch(startProfileEdit);
```
right here (before dispatching the validation updates)